File: src/main/java/io/github/willslaunwhite/array/binarysearch/BinarySearch.java
package io.github.willslaunwhite.array.binarysearch;

public class BinarySearch {
   public int binarySearch(int[] nums, int target) {
      for (int left = 0, right = nums.length - 1; left <= right;) {
         int middle = (left + right) / 2;
         if (nums[middle] < target) {
            left = middle + 1;
         } else if (nums[middle] > target) {
            right = middle - 1;
         } else {
            return middle;
         }
      }

      return -1;
   }
}

File: python/binary_search/problems/binary_search.py
def binary_search(nums: list[int], target: int) -> int:
    left: int = 0
    right: int = len(nums) - 1
    while left <= right:
        middle: int = (right + left) // 2
        if nums[middle] < target:
            left = middle + 1
        elif nums[middle] > target:
            right = middle - 1
        else:
            return middle
    return -1
